src/celeritas/optical/model/AbsorptionModel.cc

```diff
@@ -0,0 +1,61 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/AbsorptionModel.cc
+//---------------------------------------------------------------------------//
+#include "AbsorptionModel.hh"
+
+#include "corecel/Assert.hh"
+#include "celeritas/io/ImportOpticalMaterial.hh"
+
+#include "../MfpBuilder.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the model from imported data.
+ */
+AbsorptionModel::AbsorptionModel(ActionId id, ImportedModelAdapter imported)
+    : Model(id, "absorption", "interact by optical absorption")
+    , imported_(std::move(imported))
```

Comment:
I was thinking we could construct these similar to our EM `Process`s:
```c++
using SPConstImported = std::shared_ptr<ImportedModels const>;
...
AbsorptionModel(ActionId, SPConstImported);
```
```suggestion
AbsorptionModel::AbsorptionModel(ActionId id, SPConstImported model_data)
    : Model(id, "absorption", "interact by optical absorption")
    , imported_(ImportModelClass::absorption, model_data)
```
unless there's a need to have the `ImportedModelAdapter` outside of the model.

Reply:
Sorry, forgot to propagate the change to the actual `AbsorptionModel`.